src/main/java/it/gov/pagopa/reward/connector/event/consumer/TransactionConsumerImpl.java

```diff
@@ -0,0 +1,80 @@
+package it.gov.pagopa.reward.connector.event.consumer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import it.gov.pagopa.common.reactive.kafka.consumer.BaseKafkaConsumer;
+import it.gov.pagopa.reward.dto.trx.RewardTransactionDTO;
+import it.gov.pagopa.reward.model.counters.UserInitiativeCounters;
+import it.gov.pagopa.reward.service.RewardErrorNotifierService;
+import it.gov.pagopa.reward.service.counters.UserInitiativeCountersUnlockMediatorService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.messaging.Message;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+@Service
+@Slf4j
+public class TransactionConsumerImpl extends BaseKafkaConsumer<RewardTransactionDTO, UserInitiativeCounters>  implements TransactionConsumer {
```

Comment:
thank you @GiovanaSolorzano. Now i find the design easier to understand than before (and much more testable).
Last thing i ask you is about the `TransactionConsumer` interface. Is it expected to be implemented by some other classes? 
If not i would remove altogether and rename the `TransactionConsumerImpl` into `TransactionConsumer`. What do you think about it?
```suggestion
public class TransactionConsumer extends BaseKafkaConsumer<RewardTransactionDTO, UserInitiativeCounters> {
```

Reply:
should not be used elsewhere. Removed the interface TransactionConsumer.